fiware-region-sanity-tests/commons/dbus_phonehome_service.py

```diff
@@ -0,0 +1,224 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2015 Telefónica Investigación y Desarrollo, S.A.U
+#
+# This file is part of FIWARE project.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+#
+# You may obtain a copy of the License at:
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For those usages not covered by the Apache version 2.0 License please
+# contact with opensource@tid.es
+
+
+import dbus
+from dbus import SystemBus
+from dbus.exceptions import DBusException
+from dbus.service import BusName
+from dbus.mainloop.glib import DBusGMainLoop
+import gobject
+from commons.constants import PHONEHOME_DBUS_NAME, PHONEHOME_TIMEOUT, PHONEHOME_SIGNAL
+import re
+
+
+class DbusPhoneHomeClient():
+
+    expected_signal_hostname = None
+    mainloop = None
+    data_received = None
+    logger = None
+
+    def __init__(self, logger):
+        """
+        Inits the DBus client and creates a new System bus
+        :param logger: Logger
+        :return: None
+        """
+
+        self.logger = logger
+        DbusPhoneHomeClient.logger = logger
+
+        self.logger.debug("Attaching to a main loop")
+        DBusGMainLoop(set_as_default=True)
+
+        self.logger.debug("Creating session in SystemBus")
+        self.bus = SystemBus()
+
+    @staticmethod
+    def timeout(mainloop, logger, *args):
+        """
+        Timeout function for DBus MainLopp
+        :param mainloop: Loop manager (MainLoop)
+        :param logger: Logger
+        :param args: Rest of arguments
+        :return: False. The function is called repeatedly until it returns FALSE,
+         at which point the timeout is automatically destroyed and the function will not be called again.
+        """
+        logger.debug("Timed out!. Aborting the wait.")
+        mainloop.quit()
+        return False
+
+    @staticmethod
+    def phonehome_signal_handler(phonehome_http_data):
+        """
+        Handler for the signal 'phonehome_signal'.
+        :param hostname: Receives the hostname of the VM emitted in the signal. If matches with the expected one,
+         main loop will be finished.
+        :return: None
+        """
+        DbusPhoneHomeClient.logger.debug("Signal received with data '%s'", phonehome_http_data)
+        hostname = re.match(".*hostname=([\w-]*)", phonehome_http_data)
+        hostname = hostname.group(1) if hostname is not None else hostname
+        DbusPhoneHomeClient.logger.debug("Received hostname: '%s'. Expected hostname: '%s'",
```

Comment:
It is both static variable in the class and instance attribute because it is used in class methods and in static class methods. If you prefer, we can use that logger as static variable in all places of ths classs (static methods and class methods) but not as global var in the module, because other class in this module could be managed with another logger instance.